legal-api/src/legal_api/services/document_meta.py

```diff
@@ -34,136 +33,236 @@ class ReportType(Enum):
+
     def get_documents(self, filing: dict):
         """Return an array of document meta for a filing."""
-        business_identifier = filing['filing']['business']['identifier']
-        filing_id = filing['filing']['header']['filingId']
-        filing_date = filing['filing']['header']['date']
-        filing_status = filing['filing']['header']['status']
-        filing_type = filing['filing']['header']['name']
-        paper_only = filing['filing']['header']['availableOnPaperOnly']
-        documents = []
+        # look up legal type
+        self._business_identifier = filing['filing']['business']['identifier']
+        business = Business.find_by_identifier(self._business_identifier)
+        if not business:
+            return []  # business not found
+        self._legal_type = business.legal_type
 
-        if filing_status not in (Filing.Status.COMPLETED.value, Filing.Status.PAID.value) or paper_only:
-            return []
+        self._filing_status = filing['filing']['header']['status']
+        is_paper_only = filing['filing']['header']['availableOnPaperOnly']
+
+        if self._filing_status not in (Filing.Status.COMPLETED.value, Filing.Status.PAID.value) or is_paper_only:
+            return []  # wrong filing status
+
+        filing_type = filing['filing']['header']['name']
+        self._filing_id = filing['filing']['header']['filingId']
+        self._filing_date = filing['filing']['header']['date']
 
         if filing_type == 'incorporationApplication':
-            documents = self.get_incorporation_application_report(filing)
-            return documents
-
-        if filing_status == Filing.Status.COMPLETED.value:
-            if filing_type == 'annualReport':
-                documents = self.get_annual_report(filing_id, business_identifier, filing_date)
-            elif filing_type == 'changeOfAddress':
-                documents = self.get_coa_report(filing_id, business_identifier, filing_date)
-            elif filing_type == 'changeOfDirectors':
-                documents = self.get_cod_report(filing_id, business_identifier, filing_date)
-            elif filing_type == 'changeOfName':
-                documents = self.get_con_report(filing_id, business_identifier, filing_date)
-            elif filing_type == 'specialResolution':
-                documents = self.get_special_resolution_report(filing_id, business_identifier, filing_date)
-            elif filing_type == 'voluntaryDissolution':
-                documents = self.get_voluntary_dissolution_report(filing_id, business_identifier, filing_date)
+            documents = self.get_incorporation_application_reports(filing)
+        elif filing_type == 'annualReport':
+            documents = self.get_ar_reports()
+        elif filing_type == 'changeOfAddress':
+            documents = self.get_coa_reports()
+        elif filing_type == 'changeOfDirectors':
+            documents = self.get_cod_reports()
+        elif filing_type == 'changeOfName':
```

Comment:
I had mentioned this in Ody's inital PR as well. These methods are basically doing the same thing with different values. I think it would be better to externalize the reports available for each filing type and status into a json so that a single method can replace all this. It would be great to have a tech debt ticket to take care of this.

Reply:
Good idea. See /bcgov/entity#4335.